### ISSUE TITLE
slf4j + logback-journal  and systemd appender

### ISSUE THREAD

**Issue by alpizano** [gnieh/logback-journal#14]
Hi , I have a project that is using slf4j and logback-journal. Everything was working perfectly, however , I recently updated one of my dependencies, com.microsoft.azure.sdk.iot:iot-device-client:1.19 (gradle), 
and then I started getting "No SLF4J Providers Found"  warning. I  solved the No providers warnings by updating my slf4j-api and slf4j-log4j12 versions to 1.8.0-beta4, but I am stilling getting a "No appenders could be found for logger (dao.hsqlmanager).
log4j:WARN Please initialize the log4j system properly"

It seems like my logback.xml file is not getting detected. Can someone please help me out?

logback.xml:
```
<configuration>
    <appender name="journal" class="org.gnieh.logback.SystemdJournalAppender">
        <encoder>
            <pattern>%m%n</pattern>
        </encoder>
    </appender>

    <root level="info">
        <appender-ref ref="journal" />
    </root>
</configuration>
```

gradle dependencies relevant to this question:

```
compile group: com.microsoft.azure.sdk.iot, name: iot-device-client, version: 1.19.1
compile group: org.slf4j, name: slf4j-api, version: 1.8.0-beta4
compile group: org.slf4j, name: slf4j-log4j12, version: 1.8.0-beta4
compile group: org.gnieh, name: logback-journal', version: 0.3.0
compileOnly group: org.projectlombok, name: lombok, version: 1.18.8
```

**Comment by satabin** [comment]
Hi,

From what I can see you are not using logback but log4j. This appender is for logback only, so you should consider dropping `slf4j-log4j` in favor of `logback-classic`.

**Comment by satabin** [comment]
Hi @alpizano did you manage to solve this issue?

**Comment by alpizano** [comment]
@satabin Hi Yes. I was able to solve this issue ,thank you very much

The issue was caused by using the iot-device-client SDK 1.19.1 from Microsoft, where they added the 1.8.x alpha version of slf4j , which causes the "No provider" and "No appender" error with our slf4j and systemd appender (pulled it in as transient dependency).

Block that 1.8.x version and using both 1.7.x for slf4j-api and log4j12 solved my issue!
